src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -736,6 +736,8 @@ void TaskSketcherElements::onSelectionChanged(const Gui::SelectionChanges& msg)
                     QSignalBlocker sigblk(ui->listWidgetElements);
                     for (int i = 0; i < ui->listWidgetElements->count(); i++) {
                         ElementItem* item = static_cast<ElementItem*>(ui->listWidgetElements->item(i));
+                        if(item->isSelected())
+                            item->setSelected(false); //if already selected, we need to reset setSelected or it won't draw subelements correctly if selecting several.

```

Comment:
This doesn't look very elegant. Couldn't the problem be fixed in `setSelected()` ?